### DISCUSSION TITLE
Ray do not work with Docker "sail up" instead of "php artisan serve"

### DISCUSSION THREAD

**Discussion by stodolapatrik** [spatie/laravel-ray#145] [General]
Hello, is there any way, how to use Ray while using Docker "sail up" instead of "php artisan serve" ?

**Comment by mikeerickson** [comment]
Have a look at the Ray documentation, it contains information on using with Docker

https://spatie.be/docs/ray/v1/environment-specific-configuration/docker

**Reply by stodolapatrik** [reply]
Thank you for your answer, this command helped me - `php artisan ray:publish-config --docker`

**Reply by rico** [reply]
@p43ck can you provide some detailed information on how you got it working?

I did the following:

1.) `sail composer require spatie/laravel-ray --dev`
2.) `php artisan ray:publish-config --docker`
3.) Set the values in `ray.php` to:

```
'host' => 'host.docker.internal',
'port' => 23517,
'remote_path' => '/var/www/html',
'local_path' => '/Users/me/projects/project',
```

4.) Added the following to `docker-compose.yml`:

```
extra_hosts:
          - "host.docker.internal:host-gateway"
```

5.) Run `sail build --no-cache && sail up -d`
6.) Started ray app (in trial mode)
6.) Added `ray('Hello');` to one of my controller methods / routes.

Unfortunately I can't see the message when calling the route.

Tested the vue package, and this is working! Same docker container ...

**Reply by rico** [reply]
Figured it out, had to add `127.0.0.1 host.docker.internal` to `/etc/hosts`. Docker desktop added `127.0.0.1 kubernetes.docker.internal`.

**Reply by StuartCreed** [reply]
Adding 127.0.0.1 host.docker.internal to /etc/hosts and adding to ray.php works for me

**Reply by vrusua** [reply]
@p43ck thank you very much! Adding 127.0.0.1 host.docker.internal to /etc/hosts on the hosting machine (MacOS in my case) solved the issue. The rest of options are exactly based on the [Ray docs for Docker](https://spatie.be/docs/ray/v1/environment-specific-configuration/docker).

@freekmurze makes it sense to add this tip (add ```127.0.0.1 host.docker.internal``` to /etc/hosts on the hosting machine) into Ray docs, or?

Cheers,
Ruslan

**Reply by freekmurze** [reply]
Feel free to PR improvement to the docs 👍
